### PR TITLE
LND support

### PR DESCRIPTION
**Pull request by justinmoon**
Builds on #1718 and #1717

So far this implements outgoing payments and tests them in cil-test.sh and has fully working tmuxinator environment, too.

TODO:

- [ ] Route hints
- [x] Subscribe HTLCs
- [ ] Intercept HTLCs

Notes:

- Split out tmuxinator windows more: 1 cli window, 1 fedimintd window, 1 lightning window with cln. + lnd + gatewayd for both, and 1 bitcoind window
- Got rid of "ln1" and "ln2" and instead calling them roughly "cln" and "lnd".
- I'm finding inding the `ILnRpcClient` very clunky because it's really intended to speak to cln-extension. Nothing like this exists for LND. I think we should consider refactoring this trait to not use cln-extension GRPC types as arguments and return types.

### CODE REVIEW

File: gateway/gateway/Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,8 @@ tokio-stream = "0.1.11"
 tonic = { version = "0.8", features = ["transport", "tls"] }
 tower-http = { version = "0.3.5", features = ["cors", "auth"] }
 url = { version = "2.3.1", features = ["serde"] }
+tonic_openssl_lnd = { git = "https://github.com/justinmoon/tonic_openssl_lnd", rev = "68b97570" }
```
Comment (okjodom) [inline gateway/gateway/Cargo.toml:52]: consider using [this fork of tonic_lnd](https://github.com/okjodom/tonic_lnd/tree/routerrpc-feature). It includes he important bits you need from `tonic_openssl_lnd`: maintained tonic ssl, i.e.
- tonic_lnd fix for ip address issue. fixed by moving to openssl
- lnd router-rpc

Additionally:
- Includes rust features, so you can pick just the LND rpc api features you need for LND support
- Is on path for proper dependency integration so we have prod ready LND integration

Comment (justinmoon) [reply gateway/gateway/Cargo.toml:52]: Nice. Will try this. Feel free to try it yourself, too.

Comment (okjodom) [reply gateway/gateway/Cargo.toml:52]: https://github.com/justinmoon/fedimint/pull/48

Comment (justinmoon) [reply gateway/gateway/Cargo.toml]: Merged

**Comment by justinmoon** [comment]
In tmuxinator, sometimes we exercise `lncli` before `lnd`'s RPC server is ready to handle commands:

```
[lncli] rpc error: code = Unknown desc = the RPC server is in the process of starting up, but not yet ready to accept calls
```

**Comment by justinmoon** [comment]
Replaced by other PRs

**Pull request closed without merging**